lib/managers/KalturaLayoutManager.js

```diff
@@ -184,14 +185,71 @@ class KalturaLayoutManager extends kaltura.KalturaManager {
 		response.dir(params);
 		if (!this.validateActionArguments(params, ['entryId', 'flavorIds'], response))
 			return;
-		const layoutManifestKey = this._getLayoutManifestKey(params);
+
+		LayoutHelper.addEntryVersion(params.entryId,
+			function(){
+				This._handleSessionIdVesrion(params, LayoutHelper.getBaseVersion(), request, response);
+			}, function(addEntryErr){
+				KalturaLogger.log(`could not add entry ${params.entryId} version to cache , due to ${util.inspect(addEntryErr)}`);
+				LayoutHelper.getEntryVersion(params.entryId,
+					function(version){
+						if(version)
+							This._handleSessionIdVesrion(params, version.id, request, response);
+						else
+							response.error(`No entry version found ,Failed to handle layout manifest`);
+					}, function(getEntryErr){
+						KalturaLogger.log(`could not get entry ${params.entryId} version from cache , due to  ${util.inspect(getEntryErr)}`);
+						response.error(`No entry version found - Failed to create layout manifest though handled : ${getEntryErr}`);
+					}
+				);
+			}
+		);
+	}
+
+	_handleSessionIdVesrion(params, versionId, request, response)
+	{
+		const This = this;
+		if(LayoutHelper.isParamDefined(params, 'sessionId'))
+		{
+			LayoutHelper.addSessionVersion(params.sessionId, { id: versionId }, KalturaConfig.config.cache.sessionIdVersionTimeout,
+				() => {
+					const layoutManifestKey = This._getLayoutManifestKey(params, versionId);
+					This._handleManifest(layoutManifestKey, request, response, params, versionId);
+				},(addSessionErr) => {
+					KalturaLogger.log(`could not add version ${versionId} to sessionId ${params.sessionId}, due to ${addSessionErr}`);
+					LayoutHelper.getSessionVersion(params.sessionId,
+						(version) => {
+							if(version)
+							{
+								const layoutManifestKey = This._getLayoutManifestKey(params, version.id);
+								This._handleManifest(layoutManifestKey, request, response, params, version.id);
+							}
+							else
+								response.error(`could not get session version from cache`);
+						},(getSessionErr) => {
+							KalturaLogger.log(`could not get version from cache to sessionId ${params.sessionId}, due to ${util.inspect(getSessionErr)}`);
+							response.error(`could not get version from cache to sessionId`);
+						}
+					);
+				});
+		}
+		else //version must be set
+		{
+			const layoutManifestKey = This._getLayoutManifestKey(params, params.version);
+			This._handleManifest(layoutManifestKey, request, response, params, params.version);
+		}
```

Comment:
add another else that states that we are missing both version and session id - should act as missing argument treatment